core/base/matrix_data.hpp

```diff
@@ -300,6 +320,69 @@ struct matrix_data {
         return res;
     }
 
+    /**
+     * Initializes a random dense matrix with a specific condition number.
+     *
+     * The matrix is generated by applying a series of random Hausholder
+     * reflectors to a diagonal matrix with diagonal entries uniformly
+     * distributed between `sqrt(condition_number)` and
+     * `1/sqrt(condition_number)`.
+     *
+     * @param size  number of rows and columns of the matrix
+     * @param condition_number  condition number of the matrix
+     * @param dist  random distribution used to generate reflectors
+     * @param engine  random engine used to generate reflectors
+     * @param num_reflectors  number of reflectors to apply from each side
+     */
+    template <typename RandomDistribution, typename RandomEngine>
+    static matrix_data cond(size_type size,
```

Comment:
I dont know if  `cond` is the right name for a function that generates a matrix.

Reply:
I agree on this point, but lack any good naming sense.

Reply:
Also, maybe templates could be documented?

Reply:
I agree on this point. Also maybe templates could be documented?

Reply:
So you call it as `matrix_data<>::cond(/*something*/)` to create it. We also do `matrix_data<>::diag(/*something*/)` to create a diagonal one,so this is kind of consistent.

I'm also not sure of a better name - maybe if we called it `with_cond(...)`, then we have:

```c++
auto m = matrix_data<>::with_cond(3, 10, ...);
```

Sorry about the doxygen brain fart - should be fixed now, along with a bunch of other documentation issues in that file.

Reply:
I am fine with both names - whatever you prefer.